### PR TITLE
Properly implement the router

### PR DESCRIPTION
**Pull request by kordianbruck**
This will make the router use normal PHP arrays which can be cached by PHP itself and reduces the overhead in finding the routes.

### CODE REVIEW

**Comment by kordianbruck** [comment]
@mammuth check if this is alright...

File: redirect/routes.php [hunk 1]
```diff
@@ -1,112 +1,86 @@
...
+    "numprog"=> [
+      "description"=> "Numerisches Programmieren",
+      "target"=> "http://www5.in.tum.de/wiki/index.php/Numerisches_Programmieren_-_Winter_15",
+      "display"=> true
+    ]
+  ];
+
+  private $sections = [
+    "Special Stuff" => [
+      'hunger', 'rooms', 'c', 'm', 'stuff','reddit', 'tedx'
```
Comment (mammuth) [inline redirect/routes.php]: Do you have any idea for avoiding duplications like this for the sections? 

Compromise: 
What do you think about grouping the routing arrays in the correct sections (like it has been in my first implementation) and iterate through the n <10 (!) sections. This is be log(n), with a n not worth naming at all...

This isn't a large scale application and the number of sections stays small; imho it's worth doing this to have it easy for contributers to update links without caring about any duplications.

**Comment by mammuth** [comment]
Nice, thanks! 
But let's see wether we get a compromise regarding the code dupes and 'performance', as commented in the file ;-)

File: redirect/routes.php (hunk 1 above)
Comment (kordianbruck) [inline redirect/routes.php]: This is the way you would build up the information in a database table. You would normally use a PK(int) for reference in order to be able to change the route, but this duplication is not so bad. Better than to have a single array which 'simulates' structure of data which have nothing in common.

**Pull request merged**